### PR TITLE
feat: add apps as subcommand

### PR DESCRIPTION
**Pull request by makram93**
Add apps as a subcommand and make arguments supported for each apps unique to them... 

- Users can now enter `jina-now start -h` to get the below help screen, where `image_to_image`, `image_to_text` and `text_to_image` are the subcommand corresponding to each app.  image

- When the user enters `jina-now start image_to_image -h`, he/she gets the help (or additional arguments) for that respective app only
- There's also an argument `--app` whose value/choices are the same as `subcommands` and is picked from `Apps` Enum. If used, the dialog to select the app will not show. Also, when using the `subcommand` the `--app` argument's value is overridden. For e.g: `jina-now start --app image_to_text image_to_image` will have the `app=image_to_image` when argument parsing is done.

### CODE REVIEW

**Comment by joschkabraun** [comment]
Great PR! I would refactor it such that all the configurations of the apps are handled within the app and not somewhere else.

File: now/cli/parser.py [hunk 1]
```diff
@@ -53,35 +54,43 @@
...
-    """Add the arguments for the jina now to the parser
-    :param parser: an optional existing parser to build upon
-    :return: the parser
-    """
+def set_img2txt_parser(parser):
+    parser.add_argument(
+        '--quality',
+        help='Choose the quality of the model that you would like to finetune',
+        type=str,
+    )
```
[the remark this answers is not in the document]
Comment (makram93) [reply now/cli/parser.py]: Thanks for the suggestion. I like it too.. Keeping all maintenance in one place.

File: now/apps/base/app.py [hunk 2]
```diff
@@ -109,6 +113,25 @@ def example_datasource(self) -> List[Datasource]:
...
 
+    def set_app_parser(self, parser, formatter) -> None:
+        """
+        This parser reads from the `options` property and parses it
+        to form the command line arguments for app
+        """
+        if self.app != '':
+            parser = parser.add_parser(
+                self.app,
+                help=self.description,
```
Comment (florian-hoenicke) [inline now/apps/base/app.py:124]: nice. I like this way of getting the attributes. Minimizes the effort of implementing the derived class

File: now/cli/parser.py [hunk 3]
```diff
@@ -17,10 +21,10 @@ def set_base_parser():
         'Help': ('💬', 'https://slack.jina.ai'),
         'Hiring!': ('🙌', 'https://career.jina.ai'),
     }
-    url_str = '\n'.join(f'- {v[0]:<10} {k:10.10}\t{v[1]}' for k, v in urls.items())
+    url_str = '\n'.join(f'- {v[0]:<15} {k:10.10}\t{v[1]}' for k, v in urls.items())
```
Comment (florian-hoenicke) [inline now/cli/parser.py:24]: What is this logic doing?

Comment (makram93) [reply now/cli/parser.py:24]: Ah this is from the core repo... It just prints the extra message at the end
image

File: now/finetuning/embeddings.py [hunk 4]
```diff
@@ -47,6 +47,6 @@ def get_encoder_config(encoder_uses: str, artifact: str) -> _ExecutorConfig:
     """
     return _ExecutorConfig(
         name='encoder',
-        uses=f'jinahub+docker://{encoder_uses}',
+        uses=f'jinahub+sandbox://{encoder_uses}',
```
[the remark this answers is not in the document]
Comment (makram93) [reply now/finetuning/embeddings.py:50]: Yes, sorry. I was working on it and by mistake, it got added to this PR. Thanks! :)

File: now/apps/base/app.py [hunk 5]
```diff
@@ -22,6 +22,10 @@ def __init__(self):
         curdir = os.path.realpath(__file__)
         self._flow_yaml = os.path.join(curdir, 'flow.yml')
 
+    @property
+    def app(self) -> str:
+        return ''
```
Comment (florian-hoenicke) [inline now/apps/base/app.py:27]: raise NotImplemented error. This is a mandatory attribute without default value

File: now/cli/parser.py [hunk 6]
```diff
@@ -99,22 +87,35 @@ def set_start_parser(sp=None):
...
+
+    sub_parser = parser.add_subparsers(
+        dest='app',
+        description='use `%(prog)-8s [sub-command] --help` '
+        'to get additional arguments to be used with each sub-command',
+        required=False,
     )
 
+    # Set parser args for the Image to Image app
+    ImageToImage().set_app_parser(sub_parser, formatter=_chf)
```
Comment (florian-hoenicke) [inline now/cli/parser.py:106]: If you add a new app, you would have to make a code change here. For instance, the MusicToMusic app would have to be enabled as you are stating in the comment.
A better way of doing it is to automatically add everything which is in the Apps list. in the constants.py

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

File: now/cli/parser.py (hunk 6 above)
Comment (makram93) [reply now/cli/parser.py:106]: Thanks for the suggestion! 👍

File: now/cli/parser.py [hunk 7]
```diff
@@ -82,24 +72,46 @@ def set_start_parser(sp=None):
...
+
+    sub_parser = parser.add_subparsers(
+        dest='app',
+        description='use `%(prog)-8s [sub-command] --help` '
+        'to get additional arguments to be used with each sub-command',
+        required=False,
+    )
+
+    # Set parser args for the enabled apps
+    for app in Apps():
```
Comment (florian-hoenicke) [inline now/cli/parser.py:112]: nice - I like

**Comment by florian-hoenicke** [comment]
Well done - that is a lean solution

**Pull request merged**